x-pack/plugins/security_solution/server/lib/detection_engine/signals/bulk_create_threshold_signals.ts

```diff
@@ -176,7 +179,10 @@ const getTransformedHits = (
 
         const source = {
           '@timestamp': get(timestampOverride ?? '@timestamp', hit._source),
-          threshold_count: docCount,
+          threshold_result: {
```

Comment:
My understanding is that these "temporary" fields (`@timestamp`, `threshold_result`) are not intended to be persisted in this format to the final signal, but instead are used to pass context from the query to the signal generation. Had you considered making this process more explicit by nesting these fields in some declarative key like `searchContext` ?

This seems indicative of some broader architectural issues: `getThresholdSignalQueryFields` seems analogous to `additionalSignalFields` except that it exists at a different level, but I could absolutely see it being moved up as `additionalSourceFields` and similarly taking from `_source.searchContext`.

Reply:
I know this isn't new code so definitely a NIT on the above; just curious what your thoughts are here.

Reply:
So I think I can address this with the signal properties merging code discussed above. `getThresholdSignalQueryFields` is going away, per our discussion last week. Those fields are just copies of the data that can be reconstructed using `signal.rule.query` and `signal.rule.filter` combined with the "searchContext" contained in `threshold_result`. It may be cumbersome to get them all into one `searchContext` object, but I'll take a look at it. The query and filter already exist in the `signal.rule` section of the signal, while `threshold_result` belongs to the signal itself (it's computed after the data are aggregated). I think this can be cleaned up a bit though, so I'll see what I can do, and then run it by you. Thanks!